src/rgw/rgw_common.h

```diff
@@ -1864,6 +1864,47 @@ static inline std::string ys_header_mangle(std::string_view name)
   return out;
 } /* ys_header_mangle */
 
+static inline std::string extract_remote_addr(req_state * const s)
+{
+  std::string remote_param = s->cct->_conf->rgw_remote_addr_param;
+  if (remote_param.empty()) {
+    remote_param = "REMOTE_ADDR";
+  }
+
+  const char *p = s->info.env->get(remote_param.c_str());
+  if (unlikely(p == nullptr)) {
+    ldpp_dout(s, 0) << "ERROR: couldn't find remote address in env" << dendl;
+
+    return "";
+  }
+
+  std::string remote_addr = p;
+
+  auto trusted_proxy_count = s->cct->_conf->rgw_trusted_proxy_count;
+  if (!trusted_proxy_count) {
+    return remote_addr;
+  }
+
+  std::vector<std::string> ips = get_str_vec(remote_addr, ",");
```

Comment:
consider using `ceph::split()` from common/split.h which returns a lazy range of string views instead of allocating a vector of strings

Reply:
@cbodley I implemented the reverse search there - as using this would lead us to walk from the zero index while we're interested in walking the other way around. But I know the overhead is negligible as we're not expecting to have many IPs in the header - So if you think for the sake of code-readability I need to change it to split, please let me know.